client/src/deploy/sign.rs

```diff
@@ -0,0 +1,56 @@
+use std::{fs, io::Write};
+
+use clap::{App, ArgMatches, SubCommand};
+
+use super::creation_common;
+use crate::{command::ClientCommand, common};
+
+use casper_node::types::Deploy;
+
+pub struct SignDeploy;
+
+impl<'a, 'b> ClientCommand<'a, 'b> for SignDeploy {
+    const NAME: &'static str = "sign-deploy";
+    const ABOUT: &'static str = "Cryptographically signs a deploy. The signature \
+    is appended to existing approvals.";
+
+    fn build(display_order: usize) -> App<'a, 'b> {
+        let subcommand = SubCommand::with_name(Self::NAME)
+            .about(Self::ABOUT)
+            .display_order(display_order)
+            .arg(common::secret_key::arg(display_order))
```

Comment:
Just noticed that you're passing the same `display_order` through to every option.  This seems to be true for all the new subcommands in this PR.

It's an understandable mistake, but the idea is that the `display_order` passed in `ClientCommand::build` is only used to define the subcommand's display order in the full app (i.e. when running `casper-client -h`).  Within each subcommand, we should define a new enum `DisplayOrder` to set the order in which that subcommands args/options are shown.  Look to existing subcommands for an example, and aim to keep the order consistent across subcommands as far as possible please.

If there's a way to improve this (e.g. using a different type for the display order of subcommands from their own args' display order, or just renaming the variable passed in `build` to e.g. `subcommand_display_order`) then feel free to make that change if you feel it'd be useful.

Reply:
Yeah, neglected to follow our previous convention. I think we should stick with our current method.